console/create_app/tests/resources/home_resource_test_api.ts

```diff
@@ -0,0 +1,119 @@
+import { assertEquals } from "../../deps.ts";
+import { server } from "../../app.ts";
+
+async function startServer() {
+  await server.run({
+    hostname: "localhost",
+    port: 1557,
+  });
+}
+
+Deno.test({
+  name: "HomeResource - GET /",
+  fn: async () => {
+    // Start the server
+    await startServer();
+
+    // Fetch a response
+    const response = await fetch("http://localhost:1557", {
+      method: "GET",
+    });
+
+    // Make assertions
+    assertEquals(response.status, 200);
+    assertEquals(
+      await response.json(),
+      {
+        message: "GET request received!",
+      },
+    );
+
+    // Stop the server
+    await server.close();
+  },
+  sanitizeResources: false,
+  sanitizeOps: false,
```

Comment:
does removing this cause issues? hoping we should avoid this especially for user facing stuff

Reply:
yea it causes leaking

Reply:
Because we import the server from app.ts which subsequently starts the main server also and never gets closed

I think it might be best we. create the server object within the test, or abstract the logic that creates the server object? then for example we import that into app.ts and. the test or something

FYI this comment applies to. all the others that are similar, i'll close all of them when this comment addressed

Reply:
8de4407 (create server_app.ts and server_web_app.ts. also export them)
5124467 (tests failed in the commit above. this commit fixes the tests)